test/src/locator.spec.ts

```diff
@@ -538,4 +538,29 @@ describe('Locator', function () {
       await expect(result).resolves.toEqual(undefined);
     });
   });
+
+  describe('Locator.prototype.filter', () => {
+    it('should work', async () => {
+      const clock = sinon.useFakeTimers({
+        shouldClearNativeTimers: true,
+      });
+      try {
+        const {page} = await getTestState();
+        page.setDefaultTimeout(5000);
+        await page.setContent(`<a>test</a>`);
+        const result = page
```

Comment:
`page.locator(':-p-text(test)').fill('abc');` should also compile. Let's add d.ts tests for those cases and also let's add tests that the right errors are thrown

Reply:
I can add d.ts tests in a separate CL to make it easier

Reply:
WDYT about doing this once we finalize the API in public? I think adding type tests right now would just be annoying for development since API changes would need to also change type tests.

Reply:
SGTM